src/next/azext_next/custom.py

```diff
@@ -307,18 +364,23 @@ def _execute_recommend_commands(cmd, rec):
 
 
 def _execute_recommend_scenarios(cmd, rec):
-    for nx_cmd in rec["nextCommandSet"]:
+    exec_idx = rec.get("executeIndex")
+    for idx, nx_cmd in enumerate(rec["nextCommandSet"]):
+        cmd_active = idx in exec_idx
+        if not cmd_active:
+            continue
```

Comment:
We may make the code simpler by traversing exec_idx first
```suggestion
    exec_idx = rec.get("executeIndex")
    for idx in exec_idx:
         nx_cmd = rec["nextCommandSet"][idx]
```